src/modules/sway/hide.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include "modules/sway/hide.hpp"
+#include <spdlog/spdlog.h>
+#include "client.hpp"
+#include "wlr-layer-shell-unstable-v1-client-protocol.h"
+
+namespace waybar::modules::sway {
+
+Hide::Hide(const std::string& id, const Bar& bar, const Json::Value& config)
+    : ALabel(config, "hide", id, "{}", 0, true), bar_(bar), windowId_(-1) {
+  ipc_.subscribe(R"(["bar_state_update","barconfig_update"])");
+  ipc_.signal_event.connect(sigc::mem_fun(*this, &Hide::onEvent));
+  bar_.window.get_style_context()->add_class("hidden");
+  // Launch worker
+  worker();
+}
+
+void Hide::onEvent(const struct Ipc::ipc_response& res) {
+  auto payload = parser_.parse(res.payload);
```

Comment:
Add a mutex here, might fix your crash

Reply:
My C++ knowledge is almost 0, I just copy pasted some bits from other places that looked correct. :disappointed: 

2253f3e0a27be97a2b9d2e1a4e987dba530846e5 crashes less, but it still fails:

```
**
Gtk:ERROR:../gtk/gtk/gtkcssinheritvalue.c:33:gtk_css_value_inherit_free: code should not be reached
**
Gtk:ERROR:../gtk/gtk/gtkcssinheritvalue.c:33:gtk_css_value_inherit_free: code should not be reached
Bail out! Gtk:ERROR:../gtk/gtk/gtkcssinheritvalue.c:33:gtk_css_value_inherit_free: code should not be reached
Bail out! Gtk:ERROR:../gtk/gtk/gtkcssinheritvalue.c:33:gtk_css_value_inherit_free: code should not be reached
Aborted (core dumped)
```
Sounds like hiding a painting bar is a no-no.

I think the mutex should be on the `setVisible` module for the bar itself, but waiting for the first paint. Is there some kind of event queue where we can add the hide/show events, instead of being done ASAP?

Reply:
Rebased this to the lastest master and force-pushed